File: 120.triangle.go
func minimumTotal(triangle [][]int) int {
    switch len(triangle) {
    case 0: return 0
    case 1: return triangle[0][0]
    }
    for i := 1; i < len(triangle); i++ {
        for j := range triangle[i] {
            var mn int
            if j == 0 {
                mn = triangle[i-1][0]
            } else if j == len(triangle[i])-1 {
                mn = triangle[i-1][len(triangle[i-1])-1]
            } else {
                mn = min(triangle[i-1][j-1], triangle[i-1][j])
            }
            triangle[i][j] += mn
        }
    }
    return min(triangle[len(triangle)-1]...)
}

func min(nums ...int) int {
    mn := nums[0]
    for i := range nums {
        if nums[i] < mn {
            mn = nums[i]
        }
    }
    return mn
}

File: 120.triangle.java
class Solution {
    public int minimumTotal(List<List<Integer>> triangle) {
        for (int level = triangle.size()-2; level >= 0; level--) {
            for (int i = 0; i < triangle.get(level).size(); i++) {
                triangle.get(level).set(
                    i, 
                    triangle.get(level).get(i) + Math.min( 
                        triangle.get(level+1).get(i), 
                        triangle.get(level+1).get(i+1)
                    )
                );
            }
        }
        return triangle.get(0).get(0);
    }
}
